core/src/main/scala/com/mesosphere/usi/core/SchedulerLogicHandler.scala

```diff
@@ -0,0 +1,146 @@
+package com.mesosphere.usi.core
+import com.mesosphere.mesos.client.MesosCalls
+import com.mesosphere.usi.core.logic.{MesosEventsLogic, SpecificationLogic}
+import com.mesosphere.usi.core.models.{PodId, SpecEvent}
+import com.typesafe.scalalogging.StrictLogging
+import org.apache.mesos.v1.scheduler.Protos.{Event => MesosEvent}
+
+/**
+  * Container class responsible for keeping track of the state and cache.
+  *
+  * ## SpecificationState vs SchedulerLogic state
+  *
+  * The SchedulerLogic has two pieces of state: SpecificationState and SchedulerState (records and statuses). The
+  * SchedulerLogic maintains the latter. All manipulation to the SchedulerLogic state is done via one of the two
+  * processes:
+  *
+  * - The specification state is updated through incoming SpecEvents (we consume and replicate the
+  * framework-implementation's specifications)
+  * - The SchedulerState (statuses, records, etc.) is updated through StateEvents that returned as intents
+  *
+  * As such, it's worth emphasizing that business logic does not have any direct side-effects, and it manipulates the
+  * SchedulerState only by returning intents. This allows us the following:
+  *
+  * - It's more efficient, since it saves us the trouble of diffing a large data-structure with each update
+  * - We have built-in guarantees that evolutions to the SchedulerState can be reliably replicated by processing these
+  * events, since they led to the changes in the first place
+  * - We can easily know which portions of the SchedulerState should be persisted during the persistence layer.
+  * - It restricts, via the type system, the portions of the state the business logic is allowed (IE: it
+  * would be illegal for the business logic to update a specification)
+  *
+  * ## Intents and Events
+  *
+  * In the SchedulerLogic code, we'll use the word intents and events. Intents are things not yet applied, and should
+  * be. Events are things that were applied and we're notifying you about.
```

Comment:
...and should be...?

Reply:
Yeah, this seems correct to me. What's unclear?